staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -166,6 +169,15 @@ func (f *FieldManager) Apply(liveObj runtime.Object, patch []byte, fieldManager
 					patchObj.GetAPIVersion(), f.groupVersion.String()))
 	}
 
+	managerField, err := NameFromObject(patchObj)
```

Comment:
It's surprising that this is done here for apply, and in the handler for other operations?

Reply:
yeah, it's because in the apply handler there is only the patch and no object to use, so the first time an object is available for getting the field on the apply path is in fieldManager.

Reply:
I suspect we can do all of that outside of fieldManager (I think it'd be more correct tbh)
```
	// Check that the patch object has the same version as the live object
	patchObj := &unstructured.Unstructured{Object: map[string]interface{}{}}

	if err := yaml.Unmarshal(patch, &patchObj.Object); err != nil {
		return nil, fmt.Errorf("error decoding YAML: %v", err)
	}
	if patchObj.GetAPIVersion() != f.groupVersion.String() {
		return nil,
			errors.NewBadRequest(
				fmt.Sprintf("Incorrect version specified in apply patch. "+
					"Specified patch version: %s, expected: %s",
					patchObj.GetAPIVersion(), f.groupVersion.String()))
	}

	managerField, err := NameFromObject(patchObj)
	if err != nil {
		return nil, fmt.Errorf("couldn't get manager field from object: %v", err)
	}
	fieldManager, err = ValidFieldManager(fieldManager, managerField)
	if err != nil || len(fieldManager) == 0 {
		return nil, fmt.Errorf("no valid fieldManager found: %v", err)
	}
```

We can move this code back into `handlers/patch.go` -> `applyPatcher applyPatchToCurrentObject`. I think it'd be overall more consistent.

Reply:
Then you can use the same method you used elsewhere to get the fieldmanager